src/deque.jl

```diff
@@ -80,6 +80,16 @@ Create a deque of type `T`.
 """
 deque(::Type{T}) where {T} = Deque{T}()
 
+# constructor for iterator
+function Deque(iter)
+    n = length(iter)
+    d = deque(eltype(iter))
```

Comment:
The `deque(::Type{T})` call should be deprecated, since we don't construct containers that way anymore. Can you change the call here:
```suggestion
    d = Deque{eltype(iter)}()
```

Reply:
Or, if you make `blksize` a parameter:
```suggestion
    d = Deque{eltype(iter)}(blksize)
```